config/routes.rb

```diff
@@ -10,9 +10,7 @@
 
   get "/map", to: "map#index"
 
-  get "/yelp", to: "yelp_test#yelp"
-
-
-
+  get  "/yelp"                , to: "yelp#yelp"   , as: :yelp
+  post "/yelp"                , to: "yelp#search"
```

Comment:
Although this uses a form to submit, this  should still be a `GET` request because this is not creating something on the server, it's still just getting data.  Will you update this?